# DSAE-Impute: Learning Discriminative Stacked Autoencoders for Imputing Single-cell RNA-seq Data

## Getting Started

DSAE-Impute can be used either on the command line or as Python package. The following instructions can help you quickly use it on your local machine.



## Method1 (Command line)

#### Installing

You can clone this directory to install the latest GitHub version:

```git
git clone https://github.com/lebhaftcode/DSAE-Impute.git
```

#### Usage

```shell
python3 ./main.py --input 'matrix.csv'  --outputdir 'results'
```

`matrix.csv`  is the true gene expression matrix with rows representing cells and columns representing gene.  `Results` is the output folder containing the main output file,  `raw.csv` and `impute.csv`.  `raw.csv` is a raw matrix that generated by mimicking the dropout events and `impute.csv` is the imputed matrix.

 ```
parser.add_argument('--input', type=str, default='data/test.csv', required=True)
parser.add_argument('--outputdir', type=str, default='data')
parser.add_argument('--dim1', type=int, default=600)
parser.add_argument('--dim2', type=int, default=256)
parser.add_argument('--epoch1', type=int, default=3000)
parser.add_argument('--epoch2', type=int, default=1000)
parser.add_argument('--learning_rate', type=float, default=4e-3)
parser.add_argument('--batch', type=int, default=64)
parser.add_argument('--print_step', type=int, default=200)
 ```

#### Running the testsï¼š 

```shell
python3 ./main.py --input 'data/test.csv'  --outputdir 'data'
```



## Method2 (Python package)

#### Installing

You can install the latest release of DSAE-Impute by using pip with the following command:

```shell
pip install DSAE-Impute
```

#### Usage

You can import the DSAE-Impute package in your python file

```python
import DSAE_Impute
input = 'matrix.csv'
outdir = 'results'
DSAE_Impute.main(input, outdir) 
```




